### PR TITLE
Spring boot data jpa solution

### PR DESCRIPTION
**Pull request by BolsunovaNataliia**

### CODE REVIEW

File: src/main/java/mate/academy/bookshop/model/Book.java [hunk 1]
```diff
@@ -26,6 +30,8 @@ public class Book {
     private BigDecimal price;
     private String description;
     private String coverImage;
+    @Column(nullable = false)
+    private boolean isDeleted = false;
 
     public Book() {
     }
```
Comment (BohdanTrue) [inline src/main/java/mate/academy/bookshop/model/Book.java]: redudant, lombok already create all constructors for you

**Review by BohdanTrue — COMMENTED** [review]
Good job! But you forgot to do one method (update) and one minor mistake

**Review by BohdanTrue — CHANGES_REQUESTED** [review]
image

File: src/main/java/mate/academy/bookshop/controller/BookController.java
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/controller/BookController.java:1]: Book Endpoints:
- PUT: /api/books/{id} (Update a specific book)

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -3,7 +3,8 @@ spring.datasource.username=root
 spring.datasource.password=x
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
 spring.jpa.properties.hibernate.dialect = org.hibernate.dialect.MySQLDialect
-spring.jpa.hibernate.ddl-auto=create-drop
+spring.jpa.hibernate.ddl-auto=validate
+changeLogFile=/db/changelog/db.changelog-master.yaml
 spring.jpa.show-sql=true
 spring.jpa.open-in-view=false
 spring.jackson.deserialization.fail-on-unknown-properties=true
```
Comment (d1sam) [inline src/main/resources/application.properties:10]:
```suggestion
spring.jackson.deserialization.fail-on-unknown-properties=true
server.servlet.context-path=/api
```

**Review by d1sam — CHANGES_REQUESTED** [review]
good job, but please look at my comments and add necessary put method to your api :)

File: src/main/java/mate/academy/bookshop/controller/BookController.java [hunk 3]
```diff
@@ -32,4 +36,16 @@ public BookDto findBookById(@PathVariable Long id) {
...
         return bookService.save(requestDto);
     }
+
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
+        bookService.deleteById(id);
+    }
+
+    @PutMapping(value = "/{id}", consumes = {"application/json"})
```
Comment (vovasalyha) [inline src/main/java/mate/academy/bookshop/controller/BookController.java:46]: Specifying `consumes` attribute is not necessary here as it is the default value when you receive some class as a body.

**Review by vovasalyha — APPROVED** [review]
LGTM

**Pull request merged**